hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/states/NodeStateMap.java

```diff
@@ -456,4 +456,16 @@ private static Predicate<DatanodeInfo> matching(NodeOperationalState state) {
   private static Predicate<DatanodeInfo> matching(NodeState health) {
     return dn -> health.equals(dn.getNodeStatus().getHealth());
   }
+
+  public void removeNode(DatanodeDetails datanodeDetails) throws NodeNotFoundException {
+    lock.writeLock().lock();
+    try {
+      UUID uuid = datanodeDetails.getUuid();
+      getNodeInfo(uuid);
```

Comment:
reason for this check? only checking if node exist or not? IMO, not requried, we can blindly remove

Reply:
removed.